destination/format/csv.go

```diff
@@ -21,26 +21,117 @@ import (
 	"encoding/json"
 	"fmt"
 	"sort"
-	"sync"
+	"strconv"
+	"time"
 
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	"github.com/go-errors/errors"
+	"github.com/hamba/avro"
 	"golang.org/x/exp/maps"
 )
 
+const YYYYMMDD = "2006-01-02"
+
+// Map between snowflake retrieved types and connector defined types
+// TODO: just create the table with the types on the left to make this simpler.
+var SnowflakeTypeMapping = map[string]string{
+	SnowflakeFixed:        SnowflakeInteger,
+	SnowflakeReal:         SnowflakeFloat,
+	SnowflakeText:         SnowflakeVarchar,
+	SnowflakeBinary:       SnowflakeBinary,
+	SnowflakeBoolean:      SnowflakeBoolean,
+	SnowflakeDate:         SnowflakeDate,
+	SnowflakeTimestampNTZ: SnowflakeTimestampNTZ,
+	SnowflakeTimestampLTZ: SnowflakeTimestampLTZ,
+	SnowflakeTimestampTZ:  SnowflakeTimestampTZ,
+	SnowflakeVariant:      SnowflakeVariant,
+	SnowflakeObject:       SnowflakeObject,
+	SnowflakeArray:        SnowflakeArray,
+	SnowflakeVector:       SnowflakeVector,
+}
+
+const (
+	// numeric types
+	SnowflakeInteger = "INTEGER"
+	SnowflakeFloat   = "FLOAT"
+	SnowflakeFixed   = "FIXED"
+	SnowflakeReal    = "REAL"
+
+	// string & binary types
+	SnowflakeText    = "TEXT"
+	SnowflakeBinary  = "BINARY"
+	SnowflakeVarchar = "VARCHAR"
+
+	// logical data types
+	SnowflakeBoolean = "BOOLEAN"
+
+	// date & time types
+	SnowflakeTimestampLTZ = "TIMESTAMP_LTZ"
+	SnowflakeTimestampNTZ = "TIMESTAMP_NTZ"
+	SnowflakeTimestampTZ  = "TIMESTAMP_TZ"
+	SnowflakeTime         = "TIME"
+	SnowflakeDate         = "DATE"
+
+	// semi-structured data types
+	SnowflakeVariant = "VARIANT"
+	SnowflakeObject  = "OBJECT"
+	SnowflakeArray   = "ARRAY"
+
+	// geospatial data types
+	SnowflakeGeography = "GEOGRAPHY"
+	SnowflakeGeometry  = "GEOMETRY"
+
+	// vector data types
+	SnowflakeVector = "VECTOR"
+)
+
+// Map from Avro Types to Snowflake Types
+// TODO: be more precise about numeric types.
+var AvroToSnowflakeType = map[string]string{
```

Comment:
```suggestion
var AvroToSnowflakeType = map[avro.Type]string{
```

and then map the simple types. `avro.Boolean: SnowflakeBoolean` potentially.